domain/modeldefaults/state/state.go

```diff
@@ -87,104 +126,137 @@ WHERE cloud_defaults.cloud_uuid = $cloudUUID.uuid
 	return rval, nil
 }
 
-// ModelCloudRegionDefaults returns the defaults associated with the model's cloud region.
-// It returns an error satisfying [modelerrors.NotFound] if the model doesn't exist.
-func (s *State) ModelCloudRegionDefaults(ctx context.Context, uuid coremodel.UUID) (map[string]string, error) {
+// ModelCloudRegionDefaults returns the defaults associated with the model's
+// cloud region. It returns an error satisfying [modelerrors.NotFound] if the
+// model doesn't exist.
+func (s *State) ModelCloudRegionDefaults(
+	ctx context.Context,
+	uuid coremodel.UUID,
+) (map[string]string, error) {
 	db, err := s.DB()
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, errors.Capture(err)
 	}
 
 	model := modelUUID{UUID: uuid.String()}
 
+	modelExistsStmt, err := s.Prepare(`
+SELECT &modelUUID.* FROM model WHERE uuid = $modelUUID.uuid
+`, model)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
 	query, err := s.Prepare(`
 SELECT (crd.key, crd.value) AS (&cloudRegionDefaultValue.*)
-FROM model m
-JOIN cloud_region cr ON cr.uuid = m.cloud_region_uuid
-LEFT JOIN cloud_region_defaults crd ON crd.region_uuid = cr.uuid
+FROM cloud_region_defaults AS crd
+JOIN cloud_region cr ON cr.uuid = crd.region_uuid
+JOIN model m ON m.cloud_region_uuid = cr.uuid
```

Comment:
This the bug fix referred to in the description right.
If a model doesn't have a region set, then no rows would be returned and we would incorrectly report model not found. At one point we filled in a (default) region for all clouds even when the user did not specify one. Seems that's no longer the case.